src/main/java/uk/gov/register/resources/RegisterCommandReader.java

```diff
@@ -47,7 +40,9 @@ public RegisterSerialisationFormat readFrom(Class<RegisterSerialisationFormat> t
 
     private RegisterSerialisationFormat parseCommands(InputStream commandStream) {
         BufferedReader buffer = new BufferedReader(new InputStreamReader(commandStream));
-        Iterator<RegisterCommand> commands = buffer.lines().map(s -> commandParser.newCommand(s)).iterator();
+        final CommandParser parser = new CommandParser();
+        buffer.lines().forEach(s -> parser.addCommand(s));
```

Comment:
CommandParser.getCommands does return an iterator. However its true that we are not streaming the input - I'm not sure if we were really doing it before either. Anyway the basic problem is that as we require add item commands to come before append entry commands, the orphan item check has to happen after all the append entry commands have been loaded.